source/slang/slang-compiler.cpp

```diff
@@ -222,6 +224,34 @@ namespace Slang
         visitor->visitEntryPoint(this, as<EntryPointSpecializationInfo>(specializationInfo));
     }
 
+    SlangResult EntryPoint::computeDependencyBasedHash(
+        slang::Checksum* outHashCode)
+    {
+        slang::Checksum hashCode;
+        MD5HashGen hashGen;
+        MD5Context context;
+        hashGen.init(&context);
+
+        assert(getName());
+        if (getName())
+        {
+            hashGen.update(&context, getName()->text);
+        }
```

Comment:
Code style: this is code is screaming for the result of `getName()` to be put into a local variable:

```
auto name = getName();
SLANG_ASSERT(name); // Note: we are supposed to be using `SLANG_ASSERT` instead of plain `assert`
if (name)
{
    hashGen.update(&context, name->text);
}
```

Reply:
It is unclear to me why this code has both an `assert` that `getName()` is non-null, but then also uses an `if` to defend against it being null.

Reply:
Leftover code that I missed going back through, done and removed